### PR TITLE
Deepseek-v3 MTP enhancements  

### PR DESCRIPTION
**Pull request by parambole**
### Description

This PR provides enhancements to the recently added Multi-Token Prediction (MTP) support. 

The key changes in this pull request include:

-   **Corrects MTP Projection Layer:** Adds `use_bias=False` to the `dense_general` projection layer within the MTP block.

-   **Fixes Positional Encoding Logic:** Ensures the internal MTP attention block receives the original, un-rolled `position_ids`.

-   **Resolves Cumulative Metric Bug:** Implements a `reset_eval_metrics()` method in `MetricLogger` and calls it from the main training loop. This prevents the `avg_mtp_acceptance_rate` and other evaluation statistics from accumulating incorrectly across multiple eval runs.

-   **Adds Documentation and Testing:** Includes a new end-to-end test script (`test_deepseek_mtp.sh`) and updates the `Run_DeepSeek.md` documentation with clear instructions on how to use the MTP feature.

-   **Fixes:** Includes fixes like standardizing the MTP input concatenation order.

### Tests

**Evaluation Metrics:** The `avg_mtp_acceptance_rate` was observed over multiple evaluation intervals and confirmed to report correctly without erroneous accumulation.

### Checklist

Before submitting this PR, please make sure (put X in square brackets):

-   [X] I have performed a self-review of my code.
-   [X] I have necessary comments in my code, particularly in hard-to-understand areas.
-   [X] I have run end-to-end tests tests and provided workload links above if applicable.
-   [X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: end_to_end/tpu/deepseek/Run_DeepSeek.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ python3 -m MaxText.train MaxText/configs/base.yml \
 
 ## Checkpoint conversion
 To get started, follow the instructions at HuggingFace ([V3](https://huggingface.co/deepseek-ai/DeepSeek-V3), [V2-Lite](https://huggingface.co/deepseek-ai/DeepSeek-V2-Lite)) to download the model. Currently, for V3, please convert it from FP8 to BF16 using script [here](https://github.com/deepseek-ai/DeepSeek-V3/blob/a878eada08ea6913f5a2ae80a43afeffdef082ef/inference/fp8_cast_bf16.py). Once downloaded and converted to BF16:
-* run [convert_deepseek_ckpt.py](../../../MaxText/convert_deepseek_ckpt.py) to convert the checkpoint for MaxText compatibility in [Orbax](https://orbax.readthedocs.io/en/latest/guides/checkpoint/orbax_checkpoint_101.html) for training and fine-tuning.
+* run [convert_deepseek_ckpt.py](../../../MaxText/convert_deepseek_ckpt.py) to convert the checkpoint for MaxText compatibility in [Orbax](https://orbax.readthedocs.io/en/latest/guides/checkpoint/orbax_checkpoint_101.html) for training and fine-tuning. When converting a checkpoint with MTP layers (like DeepSeek-V3), be sure to add the --enable_mtp flag to process them correctly.
```
Comment (RissyRan) [inline end_to_end/tpu/deepseek/Run_DeepSeek.md]: Nit `enable_mtp` with single quote.

File: end_to_end/tpu/deepseek/Run_DeepSeek.md [hunk 2]
```diff
@@ -84,6 +84,27 @@ python3 -m MaxText.train MaxText/configs/base.yml \
     ici_fsdp_parallelism=1
 ```
 
+### Fine-tuning with MTP on v5p-256
```
Comment (RissyRan) [inline end_to_end/tpu/deepseek/Run_DeepSeek.md]: Shall we use the same format? https://screenshot.googleplex.com/C2HyAwxua8t8NQF Or adjust other 2 examples with same format?

File: end_to_end/tpu/deepseek/v3-671b/test_deepseek_mtp.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    steps=10000 \
+    max_target_length=2048 \
+    tokenizer_type=huggingface \
+    tokenizer_path=${TOKENIZER_PATH} \
+    attention=flash \
+    dtype=bfloat16 \
+    weight_dtype=bfloat16 \
+    enable_checkpointing=true \
+    mtp_num_layers=1 \
+    mtp_loss_scaling_factor=0.1
```
Comment (RissyRan) [inline end_to_end/tpu/deepseek/v3-671b/test_deepseek_mtp.sh]: Nit: extra line

**Review by RissyRan — APPROVED** [review]
LGTM! Just a few minor comments.

**Review by gobbleturk — APPROVED** [review]

File: end_to_end/tpu/deepseek/Run_DeepSeek.md (hunk 1 above)
Comment (parambole) [reply end_to_end/tpu/deepseek/Run_DeepSeek.md]: Done.

File: end_to_end/tpu/deepseek/Run_DeepSeek.md (hunk 2 above)
Comment (parambole) [reply end_to_end/tpu/deepseek/Run_DeepSeek.md]: Done.

File: end_to_end/tpu/deepseek/v3-671b/test_deepseek_mtp.sh (hunk 3 above)
Comment (parambole) [reply end_to_end/tpu/deepseek/v3-671b/test_deepseek_mtp.sh]: Done.
